porcupine/plugins/langserver.py

```diff
@@ -573,17 +564,14 @@ def forget_tab(self, tab: tabs.FileTab, *, may_shutdown: bool = True) -> None:
                 # it was never fully started
                 self._process.kill()
 
-    def request_completions(self, event: utils.EventWithData) -> None:
+    def request_completions(self, tab: tabs.FileTab, event: utils.EventWithData) -> None:
         if self._lsp_client.state != lsp.ClientState.NORMAL:
             self.log.warning(
                 f"autocompletions requested but langserver state == {self._lsp_client.state!r}"
             )
             return
 
-        tab = event.widget
-        assert isinstance(tab, tabs.FileTab) and tab.path is not None
```

Comment:
Presumably `tab.path is not None` still needs to be asserted to make mypy happy.